### DISCUSSION TITLE
Why OpenTelemetry does not recommend logging to console or stdout?

### DISCUSSION THREAD

**Discussion by POnakS** [open-telemetry/opentelemetry-dotnet#5389] [Q&A] [answered]
Hello, I see following message on [console log](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Exporter.Console/README.md) exporter:

```
This exporter is intended to be used during learning how telemetry data are created and exported. It is not recommended for any production environment.
```

Best practice related to logging is to log to stdout.

I see following downsides related to non-stdout logging:
* if something is misconfigured you see nothing - stdout is always there
* you need different setup for local enviornment (you don't have otel exporter in your terminal)

My logs could be collected on node and then sent to opentelemetry collector via infrastructure component.

OpenTelemetry .Net Sdk takes different approach - what is the motivation behind that?

**Comment by cijothomas** [comment, marked as answer]
> Why OpenTelemetry does not recommend logging to console or stdout?

I believe this is more like a question to the OpenTelemetry spec, rather than a question to OpenTelemetry .NET implementation. OTel spec has as stdout exporter, but the spec never made an opinion or specification for how the output should be. Each languages follow their own format, and there is no guarantee that each language will not change its output format. This makes it difficult/impossible for one to use this for any automatic processing. Also see https://github.com/open-telemetry/opentelemetry-dotnet/issues/5372

> Best practice related to logging is to log to stdout.

This maybe true in certain scenarios, but not something OTel spec has recommended. Infact, the [default exporter](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/sdk-environment-variables.md#exporter-selection) for OTel instrumented apps is the OTLP Exporter. An OTel Collector could of course output to stdout.

OTel .NET's console exporter is paired with simpleprocessors (this was the case even before [spec clarified](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk_exporters/stdout.md) that this should be the case!) and is highly unlikely that such a system will perform well for production workloads.

**Reply by POnakS** [reply]
Thanks, this explains a lot. Do you have any experience of using 3rd party logging libraries (i.e. NLog, Serilog) with OpenTelemetry?

**Comment by POnakS** [comment]
There is File Exporter protocol:
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/file-exporter.md#json-file-serialization
which contains specification of potential stdout format for open telemetry logs.
